scaling/plugins/sge.py

```diff
@@ -0,0 +1,46 @@
+import subprocess
+import shlex
+import logging
+import json
+
+log = logging.getLogger(__name__)
+
+# get nodes requested from pending jobs
+def pending(instance_type):
+    command = "qstat -g d -s p -u '*'"
+    slots = run_command(command)
+    return nodes(slots, instance_type)
+
+# get nodes requested from pending jobs
+def running(instance_type):
+    command = "qstat -g d -s r -u '*'"
+    slots = run_command(command)
+    return nodes(slots, instance_type)
+
+def nodes(slots, instance_type):
+    if slots <= 0:
+        return 0
+    with open('/opt/cfncluster/instances.json') as f:
+        instances = json.load(f)
+        vcpus = int(instances[instance_type]["vcpus"])
+        log.info("Instance %s has %s slots." % (instance_type, vcpus))
+        return -(-slots // vcpus)
+
+
+def run_command(command):
+    args = shlex.split(command)
+    pending = 0
+    try:
+       _output = subprocess.check_output(args)
```

Comment:
Alas `check_output` was introduced in Python 2.7 and this is not going to work on Centos6 with Python 2.6.6.
We could make an helper module to cope with this. See also PR https://github.com/awslabs/cfncluster-node/pull/32

Reply:
ah yes, I can change this function to use subprocess.Popen